src/guest_config/src/guest_config_compiler_bin.rs

```diff
@@ -0,0 +1,366 @@
+
+    use super::{
+        build_arg_parser, compile, get_argument_values, Arguments, GuestConfigCompilerError,
+        GUESTCONFIG_DEFAULT_OUTPUT_FILENAME,
+    };
+    use crate::GUESTCONFIG_COMPILER_FILE_NAME;
+
+    #[test]
+    fn test_errors() {
+        let path = "nonexistent";
+        assert_eq!(
+            format!(
+                "{:?}",
+                GuestConfigCompilerError::IOError(
+                    String::from(path),
+                    io::Error::from_raw_os_error(2)
+                )
+            ),
+            r#"IOError("nonexistent", Os { code: 2, kind: NotFound, message: "No such file or directory" })"#
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                GuestConfigCompilerError::TemplateConfiguration(GuestConfigurationError::CpuId)
+            ),
+            format!(
+                "{}",
+                r#"Error processing virtualized guest's configuration. See error [Failed to configure CPU features via CPUID]"#
+            )
+        );
+    }
+
+    #[test]
+    fn test_get_arguments() {
+        let arg_parser = build_arg_parser();
+        // correct arguments
+        let arguments = &mut arg_parser.arguments().clone();
+        arguments
+            .parse(
+                vec![GUESTCONFIG_COMPILER_FILE_NAME, "--input-file", "foo.txt"]
+                    .into_iter()
+                    .map(String::from)
+                    .collect::<Vec<String>>()
+                    .as_ref(),
+            )
+            .unwrap();
+        assert_eq!(
+            get_argument_values(arguments).unwrap(),
+            Arguments {
+                input_file: "foo.txt".to_string(),
+                output_file: GUESTCONFIG_DEFAULT_OUTPUT_FILENAME.to_string(),
+            }
+        );
+
+        let arguments = &mut arg_parser.arguments().clone();
+        arguments
+            .parse(
+                vec![
+                    GUESTCONFIG_COMPILER_FILE_NAME,
+                    "--input-file",
+                    "foo.txt",
+                    "--output-file",
+                    "/path.to/file.txt",
+                ]
+                .into_iter()
+                .map(String::from)
+                .collect::<Vec<String>>()
+                .as_ref(),
+            )
+            .unwrap();
+        assert_eq!(
+            get_argument_values(arguments).unwrap(),
+            Arguments {
+                input_file: "foo.txt".to_string(),
+                output_file: "/path.to/file.txt".to_string(),
+            }
+        );
+
+        // missing --input-file
+        let arguments = &mut arg_parser.arguments().clone();
+        assert!(arguments
+            .parse(
+                vec![GUESTCONFIG_COMPILER_FILE_NAME]
+                    .into_iter()
+                    .map(String::from)
+                    .collect::<Vec<String>>()
+                    .as_ref(),
+            )
+            .is_err());
```

Comment:
It is usually better to match on the specific error type.
```rust
assert_eq!(my_result, Err(/* ... */));
```
or if the error type doesn't implement `std::cmp::Eq`:
```rust
assert!(matches!(my_result, Err(/* ... */) if /* ... */ ));
```